### ISSUE TITLE
FatalException on Electron 19

### ISSUE THREAD

**Issue by semireg** [node-usb/node-usb#560]
I can reproduce a fatal exception on Electron 19 by timing the unplugging of a USB device with a USB read. This is on macOS 12.4 using Electron 19.1.7 with node-usb 2.5.2.

```
 1: 0x10ea6c3b8 node::FatalException(v8::Isolate*, v8::TryCatch const&) [/path/to/app/node_modules/electron/dist/Electron.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
 2: 0x10ea6c538 node::OnFatalError(char const*, char const*) [/path/to/app/node_modules/electron/dist/Electron.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
 3: 0x10ea6c3d0 node::OnFatalError(char const*, char const*) [/path/to/app/node_modules/electron/dist/Electron.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
 4: 0x10ea25764 napi_open_callback_scope [/path/to/app/node_modules/electron/dist/Electron.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
 5: 0x101c3d1e8 Napi::Error::Error(napi_env__*, napi_value__*) [/path/to/app/app/node_modules/usb/build/Release/usb_bindings.node]
 6: 0x101c49d74 handleCompletion(Transfer*) [/path/to/app/app/node_modules/usb/build/Release/usb_bindings.node]
 7: 0x101c3e460 Napi::ThreadSafeFunction::CallJS(napi_env__*, napi_value__*, void*, void*) [/path/to/app/app/node_modules/usb/build/Release/usb_bindings.node]
 8: 0x10ea28048 node_api_get_module_file_name [/path/to/app/node_modules/electron/dist/Electron.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
 9: 0x108e5dc28 uv_async_send [/path/to/app/node_modules/electron/dist/Electron.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
10: 0x108e6ebc4 uv_free_interface_addresses [/path/to/app/node_modules/electron/dist/Electron.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
11: 0x108e5e034 uv_run [/path/to/app/node_modules/electron/dist/Electron.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
12: 0x108faba4c node::FreeArrayBufferAllocator(node::ArrayBufferAllocator*) [/path/to/app/node_modules/electron/dist/Electron.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework]
```

I'm happy to help test. Just let me know!

**Comment by strazzere** [comment]
Try the latest release, this should be fixed now as I believe it is linked to #537 #548 via PR #556.

**Comment by semireg** [comment]
Thank you. I've upgraded to this version and will report/reopen if there are any issues. 👍
